drivers/adc/adc_stm32.c

```diff
@@ -862,25 +862,33 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 		if ((__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_TEMPSENSOR_ADC1) == channel_id)
 		    && (config->base == ADC1)) {
 			adc_stm32_set_common_path(dev, LL_ADC_PATH_INTERNAL_TEMPSENSOR);
+			k_sleep(K_USEC(LL_ADC_DELAY_TEMPSENSOR_STAB_US));
```

Comment:
havent check the files, but I wonder if this need to be guarded by `#ifdef LL_ADC_DELAY_TEMPSENSOR_STAB_US` as well?

Reply:
This part of the code is specific to the STM32G4x SoC and they all have that constant defined.